path.go

```diff
@@ -528,8 +528,20 @@ func (n *selectorNode) filter(node ast.Node) (ast.Node, error) {
 			}
 			return filtered, nil
 		}
+	case ast.AnchorType:
+		value := node.(*ast.AnchorNode)
+		if n.child == nil {
+			return n.filter(value.Value)
+		}
+		return n.child.filter(value.Value)
```

Comment:
I don't know if it could ever reach this, at least not with the current unit test. @goccy could you help me come up with a test case that would reach this? Then I can make sure it will work with this as well.